ui/tract-renderer/src/main/java/org/cru/godtools/tract/ui/controller/TabsController.kt

```diff
@@ -0,0 +1,106 @@
+package org.cru.godtools.tract.ui.controller
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.annotation.CallSuper
+import androidx.annotation.UiThread
+import com.google.android.material.tabs.TabLayout
+import com.google.android.material.tabs.TabLayout.OnTabSelectedListener
+import org.ccci.gto.android.common.material.tabs.setBackgroundTint
+import org.cru.godtools.base.model.Event
+import org.cru.godtools.tract.databinding.TractContentTabsBinding
+import org.cru.godtools.tract.viewmodel.BaseViewHolder
+import org.cru.godtools.xml.model.Tab
+import org.cru.godtools.xml.model.Tabs
+import org.cru.godtools.xml.model.primaryColor
+
+class TabsController private constructor(
+    private val binding: TractContentTabsBinding,
+    parentViewHolder: BaseViewHolder<*>?
+) : BaseViewHolder<Tabs>(Tabs::class.java, binding.root, parentViewHolder), OnTabSelectedListener {
+    internal constructor(parent: ViewGroup, parentViewHolder: BaseViewHolder<*>?) :
+        this(TractContentTabsBinding.inflate(LayoutInflater.from(parent.context), parent, false), parentViewHolder)
+
+    init {
+        binding.tabs.addOnTabSelectedListener(this)
+    }
+
+    private val tabCache = UiControllerCache(binding.tab, this)
+
+    private var bindingTabs = false
+    private var tabControllers = emptyList<TabController>()
+
+    // region Lifecycle
+    @UiThread
+    override fun onBind() {
+        super.onBind()
+        binding.model = model
+        bindTabs()
+    }
+
+    @CallSuper
+    override fun onContentEvent(event: Event) {
+        super.onContentEvent(event)
+        checkForTabEvent(event)
+        propagateEventToChildren(event)
+    }
+
+    override fun onTabSelected(tab: TabLayout.Tab) {
+        val holder = showTabContent(tab.position) ?: return
+        if (!bindingTabs) holder.trackSelectedAnalyticsEvents()
```

Comment:
Curious on style here. I know for Java, we made the [decision](https://docs.google.com/document/d/16TGhx6nL3d3cjN8Rmpo-oY55ErTSUT6sZx8RVDTpgJU/edit) (department-wide) not to inline conditional statements. Is there a reason to inline them for Kotlin?

Reply:
either syntax is valid, and there are not any official Kotlin styles one way or the other that I'm aware of

I've been preferring one line if statements when it's simple logic to help consolidate the code.

Reply:
I think at the core I'm viewing Kotlin as being more similar to script languages than Java, and script languages tend to do one line if's vs blocks